Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/AbstractPatientDiscoveryAuditTransforms.java

```diff
@@ -0,0 +1,160 @@
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_ROLE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_DISPLAY_NAME);
+
+        if (aa != null && patientId != null && !aa.isEmpty() && !patientId.isEmpty()) {
+            participantObject.setParticipantObjectID(createPatientId(aa, patientId));
+        }
+
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+
+        return auditMsg;
+    }
+
+    protected static String createPatientId(String assigningAuthId, String patientId) {
+        LOG.info("createPatientId()");
+        return patientId + "^^^&" + assigningAuthId + "&ISO";
+    }
+
+    protected AuditMessageType getQueryParamsParticipantObjectIdentificationForRequest(PRPAIN201305UV02 request,
+        AuditMessageType auditMsg) throws JAXBException {
+        LOG.info("getQueryParamsParticipantObjectIdentificationForRequest()");
+        ParticipantObjectIdentificationType participantObject = buildBaseParticipantObjectIdentificationType(
+            PRPAIN201305UV02Parser.getQueryId(request));
+        participantObject.setParticipantObjectQuery(getParticipantObjectQueryForRequest(request));
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+        return auditMsg;
+    }
+
+    protected byte[] getParticipantObjectQueryForRequest(PRPAIN201305UV02 request) throws JAXBException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        if (request.getControlActProcess() != null && request.getControlActProcess().getQueryByParameter() != null) {
+            getMarshaller().marshal(request.getControlActProcess().getQueryByParameter(), baos);
+        }
+        return baos.toByteArray();
+    }
+
+    protected ParticipantObjectIdentificationType buildBaseParticipantObjectIdentificationType(
+        String participantObjectId) {
+        LOG.info("buildBaseParticipantObjectIdentificationType()");
+        ParticipantObjectIdentificationType participantObject = createParticipantObjectIdentification(
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_QUERYPARAMS_OBJ_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_QUERYPARAMS_OBJ_TYPE_CODE_ROLE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_CODE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_DISPLAY_NAME);
+        participantObject.setParticipantObjectID(participantObjectId);
+        return participantObject;
+    }
+
```

Comment:
The functionality that's same across Patient Discovery and Patient Discovery Deferred Response can also be refactored and defined in abstract class. It is missing.